packages/remix-dev/compiler/js/compiler.ts

```diff
@@ -156,11 +149,12 @@ export const create = async (
 ): Promise<Compiler> => {
   let compiler = await esbuild.context({
     ...createEsbuildConfig(ctx, refs),
+    write: false,
```

Comment:
We need to manually write the result of the browser build to disk so that [esbuild-plugins-node-modules-polyfill](https://github.com/imranbarbhuiya/esbuild-plugins-node-modules-polyfill) can check the build output in memory for any missing/unconfigured polyfills after tree-shaking has occured. This is important in the context of Remix since our route files often depend on Node builtins. If `write` is enabled, the `outputFiles` are not kept in memory and are not passed to the `onEnd` callback of the esbuild plugin.

If you're interested, here's the PR where I implemented this: https://github.com/imranbarbhuiya/esbuild-plugins-node-modules-polyfill/pull/152